wicket-core/src/main/java/org/apache/wicket/markup/html/form/Form.java

```diff
@@ -1266,58 +1250,44 @@ public void component(final FormComponent<?> formComponent, IVisit<Void> visit)
-	 * render an extra field with an invisible style so that pressing enter in one of the textfields
-	 * will do a form submit using this component. This method is overridable as what we do is best
-	 * effort only, and may not what you want in specific situations. So if you have specific
-	 * usability concerns, or want to follow another strategy, you may override this method.
+	 * If a default IFormSubmittingComponent was set on this form, this method will be called and
+	 * can be used to render extra items to HTML.
 	 *
 	 * @see #addDefaultSubmitButtonHandler(IHeaderResponse)
 	 */
+	@Deprecated
 	protected void appendDefaultButtonField()
 	{
-		AppendingStringBuffer buffer = new AppendingStringBuffer();
-
-		// hidden div
-		buffer.append(String.format("<div hidden=\"\" class=\"%s\">",
-			getString(HIDDEN_FIELDS_CSS_CLASS_KEY)));
-
-		// add an empty textfield (otherwise IE doesn't work)
-		buffer.append("<input type=\"text\" tabindex=\"-1\" autocomplete=\"off\"/>");
-
-		// add the submitting component
-		buffer
-			.append(String.format("<input id=\"%s\" type=\"submit\" tabindex=\"-1\" name=\"%s\" />",
-				getHiddenFieldsId(HIDDEN_FIELDS_SUBMIT_IDX),
-				defaultSubmittingComponent.getInputName()));
-
-		// close div
-		buffer.append("</div>");
-
-		getResponse().write(buffer);
+		// intentionally left empty for backward compatibility
 	}
 
 	/**
-	 * Where {@link #appendDefaultButtonField()} renders the markup for default submit button
-	 * handling, this method attaches the event handler to its 'click' event. The 'click' event on
-	 * the hidden submit button will be dispatched to the selected default submit button. As with
-	 * {@link #appendDefaultButtonField()} this method can be overridden when the generated code
-	 * needs to be adjusted for a specific usecase.
+	 * This method attaches the event handler to its 'enter' event.
 	 *
 	 * @param headerResponse
 	 *            The header response.
 	 */
 	protected void addDefaultSubmitButtonHandler(IHeaderResponse headerResponse)
 	{
-		final Component submittingComponent = (Component) defaultSubmittingComponent;
-		AppendingStringBuffer buffer = new AppendingStringBuffer();
-		buffer.append("var b=document.getElementById('");
-		buffer.append(submittingComponent.getMarkupId());
-		buffer.append("'); if (b!=null && b.onclick!=null && typeof(b.onclick) != 'undefined') ");
-		buffer.append(
-			"{  var r = Wicket.bind(b.onclick, b)(); if (r != false) b.click(); } else { b.click(); };  return false;");
-		headerResponse.render(OnEventHeaderItem
-			.forMarkupId(getHiddenFieldsId(HIDDEN_FIELDS_SUBMIT_IDX), "click", buffer.toString()));
+		final Component component = (Component) defaultSubmittingComponent;
+		String submitId = component.getMarkupId();
+
+		AppendingStringBuffer script = new AppendingStringBuffer();
+		script.append("if ($('#").append(submitId).append("').is(':hidden')) return;");
```

Comment:
Please do this without jQuery. E.g. with Wicket.** APIs or vanilla JS. 
A developer should be able to replace wicket-ajax-jquery.js with a custom impl and get rid of jQuery.